apps/web/src/components/templates/notification-setting-form/TemplatePreference.tsx

```diff
@@ -0,0 +1,114 @@
+import { Divider, Grid, InputWrapper } from '@mantine/core';
+import { useEnvController } from '../../../store/use-env-controller';
+import { inputStyles } from '../../../design-system/config/inputs.styles';
+import { Checkbox, colors, Text } from '../../../design-system';
+import { Controller, useFormContext } from 'react-hook-form';
+import styled from 'styled-components';
+import { channels } from '../../../pages/templates/shared/channels';
+
+export function TemplatePreference() {
+  return (
+    <>
+      <Divider
+        label={<Text color={colors.B40}>Your user preferences</Text>}
+        color={colors.B30}
+        my="xs"
+        labelPosition="center"
+      />
+
+      <Grid>
+        <Grid.Col span={6}>
+          <ChannelPreference />
+        </Grid.Col>
+        <Grid.Col span={6}>
+          <CriticalPreference />
+        </Grid.Col>
+      </Grid>
+    </>
+  );
+}
+
+export function ChannelPreference() {
+  const { control } = useFormContext();
+
+  return (
+    <Controller
+      name="preferenceSettings"
+      control={control}
+      render={({ field }) => {
+        const { readonly } = useEnvController();
+        const preferences = field.value;
+        const mock = { channel: true };
+        const data = preferences ? preferences : mock;
```

Comment:
Hey, guys @ainouzgali @scopsy so here I added a 'mock' just in order to present something before the fetch is rendered. what do you think about this one? should remove it, keep it, expand it to all channels or add a skeleton?